synapse/lib/storm.py

```diff
@@ -4321,6 +4321,67 @@ async def execStormCmd(self, runt, genr):
                 await self.delEdges(node, verb, n2)
                 yield node, path
 
+class OnceCmd(Cmd):
+    '''
+    Ensure a named action only runs on a node once.
+
+    For example, update the seen times on a set of files just once:
+
+        file:bytes#my.files | once node:seen | [ +.seen=$lib.time.now() ]
+
+    If you insert the once command with the same name on the same nodes, they will be
+    dropped from the pipeline. So in the above example, if we run it again, none of
+    the seen times will be updated, and all the nodes will be dropped from the pipeline.
+
+    So this:
+
+        file:bytes#my.files | once node:seen
+
+    Yields no nodes.
+
+    The once command utilizes a node's nodedata cache, and you can use the --asof parameter
+    to update the named action's timestamp in order to bypass/update the once timestamp. So
+    this command:
+
+        inet:ipv4#my.addresses | once node:enrich --asof now | my.enrich.command
+
+    Will yield all the enriched nodes the first time around. The second time that command is
+    run, all of those nodes will be re-enriched, as the asof timestamp will be greater the
+    second time around, so no nodes will be dropped.
+    '''
+    name = 'once'
+
+    def getArgParser(self):
+        pars = Cmd.getArgParser(self)
+        pars.add_argument('name', help='Name of the action to only perform once')
+        pars.add_argument('--asof', default='', help='')
```

Comment:
we can make this a `time='time'` to have the valu automatically normed if set

Reply:
I think this might have meant to be `type='time'` :D